Source/carthage/Bootstrap.swift

```diff
@@ -26,7 +26,19 @@ public struct BootstrapCommand: CommandType {
 					carthage.println(formatting.bullets + "No Cartfile.resolved found, updating dependencies")
 					return project.updateDependencies(shouldCheckout: options.checkoutAfterUpdate)
 				}
-
+				
+				if let depsToUpdate = options.dependenciesToUpdate {
+					let resolvedCartfileContents = try! String(contentsOfURL: project.resolvedCartfileURL, encoding: NSUTF8StringEncoding)
+					let resolvedCartfile = ResolvedCartfile.fromString(resolvedCartfileContents).value!
```

Comment:
There's a few ways to do it, but I'd probably do it like this:

```swift
let checkDependencies: SignalProducer<(), CarthageError>
if let depsToUpdate = options.dependenciesToUpdate {
    checkDependencies = project
        .loadRespolvedDependencies()
        .flatMap(.Concat) { cartfile in
            if … {
                return .error(…)
            } else {
                return .empty()
            }
        }
} else {
    checkDependencies = .empty()
}

let checkoutDependencies: SignalProducer<(), CarthageError>
if options.checkoutAfterUpdate {
    checkoutDependencies = project.checkoutResolvedDependencies(options.dependenciesToUpdate)
} else {
    checkoutDependencies = .empty
}

return checkDependencies.then(checkoutDependencies)
```

⚠️ _I wrote this directly into my browser, so it may have errors._